File: android/src/main/java/com/example/mobile_number/SimCard.java
package com.example.mobile_number;

import android.os.Build;
import android.telephony.SubscriptionInfo;

import androidx.annotation.RequiresApi;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Map;

public class SimCard {

    private String carrierName;
    private  String displayName;
    private int slotIndex;
    private String number;
    private String countryIso;
    private  String countryPhonePrefix;

    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP_MR1)
    public SimCard(SubscriptionInfo subscriptionInfo) {
        this.carrierName = subscriptionInfo.getCarrierName().toString();
        this.displayName =subscriptionInfo.getDisplayName().toString();
        this.slotIndex = subscriptionInfo.getSimSlotIndex();
        this.number = subscriptionInfo.getNumber();
        this.countryIso = subscriptionInfo.getCountryIso();
        this.countryPhonePrefix = CountryToPhonePrefix.prefixFor(subscriptionInfo.getCountryIso());
    }

    // final JSONArray jsonArray = new JSONArray();

    JSONObject toJSON(){
        JSONObject json = new JSONObject();
        try {
            json.put("carrierName", carrierName);
            json.put("displayName", displayName);
            json.put("slotIndex", slotIndex);
            json.put("number", number);
            json.put("countryIso", countryIso);
            json.put("countryPhonePrefix", countryPhonePrefix);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return json;

    }
}

File: lib/sim_card.dart
import 'dart:convert';

class SimCard {
  final String? carrierName;
  final String? displayName;
  final int? slotIndex;
  final String? number;
  final String? countryIso;
  final String? countryPhonePrefix;

  SimCard({
    this.carrierName,
    this.displayName,
    this.slotIndex,
    this.number,
    this.countryIso,
    this.countryPhonePrefix,
  });

  factory SimCard.fromMap(Map<String, dynamic> json) => SimCard(
        carrierName: json["carrierName"],
        displayName: json["displayName"],
        slotIndex: json["slotIndex"],
        number: json["number"],
        countryIso: json["countryIso"],
        countryPhonePrefix: json["countryPhonePrefix"],
      );

  Map<String, dynamic> toMap() => {
        "carrierName": carrierName,
        "displayName": displayName,
        "slotIndex": slotIndex,
        "number": number,
        "countryIso": countryIso,
        "countryPhonePrefix": countryPhonePrefix,
      };

  static List<SimCard> parseSimCards(String str) =>
      List<SimCard>.from(json.decode(str).map((x) => SimCard.fromMap(x)));

  static String simCardToJson(List<SimCard> data) =>
      json.encode(List<dynamic>.from(data.map((x) => x.toMap())));
}
